edgelet/edgelet-kube/src/runtime.rs

```diff
@@ -183,46 +183,55 @@ where
             name: String,
             nodes_count: u32,
         };
-
-        let fut = self
-            .client
-            .lock()
-            .expect("Unexpected lock error")
-            .borrow_mut()
-            .list_nodes()
-            .map_err(|err| {
-                Error::from(err.context(ErrorKind::RuntimeOperation(RuntimeOperation::SystemInfo)))
-            })
-            .map(|nodes| {
-                // Accumulate the architectures and their node counts into a map
-                let architectures = nodes
-                    .items
-                    .into_iter()
-                    .filter_map(|node| {
-                        node.status
-                            .and_then(|status| status.node_info.map(|info| info.architecture))
-                    })
-                    .fold(HashMap::new(), |mut architectures, current_arch| {
-                        let count = architectures.entry(current_arch).or_insert(0);
-                        *count += 1;
-                        architectures
-                    });
-
-                // Convert a map to a list of architectures
-                let architectures = architectures
-                    .into_iter()
-                    .map(|(name, count)| Architecture {
-                        name,
-                        nodes_count: count,
+        let fut = if self.settings.has_cluster_rbac() {
```

Comment:
As mentioned in the helm chart changes, we should aim to query what we can without the added complexity of settings. The existence of the role is sufficient to communicate the operator is interested in grabbing system information.